src/app/components/elements/views/ActivityGraph.tsx

```diff
@@ -1,18 +1,24 @@
-import React, { useEffect } from "react";
-import { bb } from "billboard.js";
+import React, { useEffect, useMemo } from "react";
+import bb, { zoom, areaSpline } from "billboard.js";
 import { AnsweredQuestionsByDate } from "../../../../IsaacApiTypes";
 import { formatISODateOnly } from "../DateString";
 
-export const ActivityGraph = ({ answeredQuestionsByDate }: { answeredQuestionsByDate: AnsweredQuestionsByDate }) => {
-  let selectedDates: string[] = [];
+const filterAndSortSelectedDates = (answeredQuestionsByDate: AnsweredQuestionsByDate) => {
   const foundDates = answeredQuestionsByDate ? Object.keys(answeredQuestionsByDate) : [];
-  if (foundDates && foundDates.length > 0) {
-    const nonZeroDates = foundDates.filter((date) => answeredQuestionsByDate && answeredQuestionsByDate[date] > 0);
-    if (nonZeroDates.length > 0) {
-      selectedDates = foundDates.sort();
-    }
+
+  if (foundDates?.some((date) => answeredQuestionsByDate[date] > 0)) {
```

Comment:
This is clearer now, although it would be even better if you saved the result of this expression in a variable. The good thing about the old `nonZeroDates` variable was that it signalled very clearly its intention.

Another thing is that you don't need the question mark, as `foundDates` will always be defined.